qdrant_client/local/local_collection.py

```diff
@@ -428,31 +478,52 @@ def recommend(
         using: Optional[str] = None,
         lookup_from_collection: Optional["LocalCollection"] = None,
         lookup_from_vector_name: Optional[str] = None,
+        strategy: Optional[types.RecommendStrategy] = None,
     ) -> List[models.ScoredPoint]:
-        search_in_vector_name, vector, query_filter = self._recommend(
+        
+        strategy = (
+            strategy if strategy is not None else types.RecommendStrategy.AVERAGE_VECTOR
+        )
+        
+        positive_vectors, negative_vectors, edited_query_filter = self._preprocess_recommend(
             positive,
             negative,
+            strategy,
             query_filter,
             using,
             lookup_from_collection,
             lookup_from_vector_name,
         )
-
+        
+        if strategy == types.RecommendStrategy.AVERAGE_VECTOR:
+            query_vector = self._recommend_average(
+                positive_vectors,
+                negative_vectors,
+            )
+        elif strategy == types.RecommendStrategy.BEST_SCORE:
+            query_vector = RecoQuery(
+            positive=positive_vectors,
+            negative=negative_vectors,
+        )
+        
+        search_in_vector_name = using if using is not None else DEFAULT_VECTOR_NAME
+        
         return self.search(
-            query_vector=(search_in_vector_name, vector),
-            query_filter=query_filter,
+            query_vector=(search_in_vector_name, query_vector),
```

Comment:
query_vector might be referenced before assignment because we don't have a raise stmt